### PR TITLE
applications: nrf5340_audio: Include revision number in the shifted hex

### PR DESCRIPTION
**Pull request by pirun**
Add revision number of PCFT build to shifted hex filename postfix like baremetal one.
Modify pcft_sign.py to find out latest shifted hex filename.

Signed-off-by: Pirun Lee <pirun.lee@nordicsemi.no>

### CODE REVIEW

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 1]
```diff
@@ -278,19 +282,46 @@ def sign(orig_pcft_hex, build_dir):
...
-                        help='Input hex file.')
-    parser.add_argument("-b", "--build_dir", required=True, type=str,
+        description='This script sign and generate netcore hex and upgradable binary for the\
+         nRF5340 Audio project on Windows and Linux')
+    parser.add_argument('-i', '--input_file', type=str,
+                        help='Input hex file name. Higher priority')
+    parser.add_argument('-b', '--build_dir', required=True, type=str,
                         help='Build folder.')
+    parser.add_argument('-I', '--input_folder', type=str,
+                        help='Input hex folder, let script choose filename automatilly.\
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:313]: automatilly = automatically

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 2]
```diff
@@ -278,19 +282,46 @@ def sign(orig_pcft_hex, build_dir):
...
+def find_hex_name(options):
+
+    in_path = Path(options.input_folder).resolve()
+
+    # Sort hex files
+    hex_files_found = sorted(in_path.glob('*.hex'))
+    pattern_found = []
+    if options.min_b0n:
+        for file in hex_files_found:
+            if re.match(ORGG_PCFT_MIN_PATTERN, file.name):
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:294]: Maybe do an inline if based on ORIG_PCFT_(...)?

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 3]
```diff
@@ -278,19 +282,46 @@ def sign(orig_pcft_hex, build_dir):
...
         formatter_class=argparse.RawDescriptionHelpFormatter,
-        description="This script sign and generate netcore hex and upgradable binary for the\
-         nRF5340 Audio project on Windows and Linux")
-    parser.add_argument('-i', '--input', required=True, type=str,
-                        help='Input hex file.')
-    parser.add_argument("-b", "--build_dir", required=True, type=str,
+        description='This script sign and generate netcore hex and upgradable binary for the\
+         nRF5340 Audio project on Windows and Linux')
+    parser.add_argument('-i', '--input_file', type=str,
+                        help='Input hex file name. Higher priority')
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:309]: "Higher priority than -I"

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 4]
```diff
@@ -278,19 +282,46 @@ def sign(orig_pcft_hex, build_dir):
...
+    parser.add_argument('-b', '--build_dir', required=True, type=str,
                         help='Build folder.')
+    parser.add_argument('-I', '--input_folder', type=str,
+                        help='Input hex folder, let script choose filename automatilly.\
+                        Lower priority')
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:314]: "Lower priority than -i"

**Review by erikrobstad — CHANGES_REQUESTED** [review]

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 5]
```diff
@@ -13,20 +13,24 @@
...
 
-PCFT_HEX = "new_fw_info_pcft.hex"
-FINAL_PCFT_HEX = "pcft_CPUNET.hex"
-FINAL_PCFT_UPDATE_BIN = "pcft_net_core_update.bin"
-ZEPHYR_BASE = os.environ["ZEPHYR_BASE"]
+PCFT_HEX = 'new_fw_info_pcft.hex'
+FINAL_PCFT_HEX = 'pcft_CPUNET.hex'
+FINAL_PCFT_UPDATE_BIN = 'pcft_net_core_update.bin'
+ORIG_PCFT_PATTERN = 'ble5-ctr-rpmsg_shifted_[0-9]'
+ORGG_PCFT_MIN_PATTERN = 'ble5-ctr-rpmsg_shifted_min_[0-9]'
```
[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Typo. Fixed.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 6]
```diff
@@ -278,19 +282,46 @@ def sign(orig_pcft_hex, build_dir):
     shutil.copy(f'{build_dir}/zephyr/net_core_app_update.bin'.format(),
                 f'{BIN_DIR}/{final_file_prefix}net_core_app_update_{fw_info_firmware_version}.bin')
 
+def find_hex_name(options):
```
[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py:288]: Fixed with autopep8.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 1 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Typo fixed.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 7]
```diff
@@ -278,19 +282,46 @@ def sign(orig_pcft_hex, build_dir):
...
+        for file in hex_files_found:
+            if re.match(ORGG_PCFT_MIN_PATTERN, file.name):
+                pattern_found.append(file)
+    else:
+        for file in hex_files_found:
+            if re.match(ORIG_PCFT_PATTERN, file.name):
+                pattern_found.append(file)
 
+    # Last hex file should be latest version.
+    return pattern_found[-1]
```
[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py:300]: Tested with python 3.8 which is the version what NCS uses.
The default [sorted()](https://docs.python.org/3.8/howto/sorting.html#sorting-basics) is ascending sort.
After sorting hex files in ascending order, we use regex to reserve the files with 4 digits build version.
So last item of pattern_found should be the largest version number.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 3 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Done.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 4 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Done.

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 2 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py]: Replace with List Comprehension style.
`pattern_found = [
    file for file in hex_files_found if re.match(ORIG_PCFT_MIN_PATTERN if options.min_b0n else ORIG_PCFT_PATTERN, file.name) is not None
]
`

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (erikrobstad) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py:300]: Got it. But isnt it better to do re.match first and then sorted()?
Then we are sure that we only sort .hex files that has the "correct controller name"

[the remark this answers is not in the document]
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py:298]: Change pattern and match whole name at once. Then we sort the result.

**Review by erikrobstad — APPROVED** [review]

**Review by alexsven — APPROVED** [review]

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py [hunk 8]
```diff
@@ -279,18 +285,42 @@ def sign(orig_pcft_hex, build_dir):
                 f'{BIN_DIR}/{final_file_prefix}net_core_app_update_{fw_info_firmware_version}.bin')
 
 
+def find_hex_name(options):
+
+    in_path = Path(options.input_folder).resolve()
+
+    # Add files only match pattern filename and 4 digits build number
+    pattern_found = sorted([
+        file for file in in_path.iterdir() if re.match(ORIG_PCFT_MIN_PATTERN if options.min_b0n else ORIG_PCFT_PATTERN, file.name) is not None
```
Comment (erikrobstad) [inline applications/nrf5340_audio/tools/buildprog/pcft_sign.py:294]: Is this line too long?

**Review by erikrobstad — APPROVED** [review]

File: applications/nrf5340_audio/tools/buildprog/pcft_sign.py (hunk 8 above)
Comment (pirun) [reply applications/nrf5340_audio/tools/buildprog/pcft_sign.py:294]: Somehow, this line passed pylint check.

**Review by rick1082 — APPROVED** [review]

**Pull request merged**